components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -952,6 +999,32 @@ def getAnnotation(self, ns=None):
             return AnnotationWrapper._wrap(self._conn, rv[0].child, link=rv[0])
         return None
 
+    def getAnnotationCounts(self, ns=None):
+        """
+        Get the annotion counts for the current object
+        """
+
+        params = omero.sys.ParametersI()
+        params.addId(long(self._oid))
+
+        q = """
+            select al from %sAnnotationLink al
+            left outer join al.parent as pa
```

Comment:
I think if you don't use parent you don't need this line , it is enough `where al.parent in (:ids)`